api/attestationconfigapi/fetcher.go

```diff
@@ -71,31 +85,39 @@ func (f *fetcher) FetchLatestVersion(ctx context.Context, variant variant.Varian
 }
 
 // fetchVersionList fetches the version list information from the config API.
-func (f *fetcher) fetchVersionList(ctx context.Context, variant variant.Variant) (List, error) {
-	fetchedList, err := apifetcher.FetchAndVerify(ctx, f.HTTPClient, f.cdnURL, List{Variant: variant}, f.verifier)
+func (f *fetcher) fetchVersionList(ctx context.Context, attestationVariant Variant) (List, error) {
+	parsedVariant, err := variant.FromString(attestationVariant.String())
+	if err != nil {
+		return List{}, fmt.Errorf("parsing variant: %w", err)
+	}
```

Comment:
So when Continuum has its own variants (and they don't exactly match what we have here in Constellation), this will break. I think this package that can only work with a certain set of variants *should* export the package specifying its variants.

Reply:
Took a look at the code
From what I can tell `variant` is only used to parse strings in this context.
We should just refactor this to work on strings instead of the custom variant type.
We will still receive appropriate error messages if objects do not exist for those strings.

We can provide constants for Constellation's supported variants, e.g. `azure-sev-snp` and `aws-sev-snp`.
I think that was intended by providing [these variables](https://github.com/edgelesssys/constellation/blob/c28e8fbd43c121017f89e28f2d4bf7b8ba796ee9/api/attestationconfigapi/fetcher.go#L21-L28)

Reply:
correct. I only kept this interface to not have to change a lot of places where this is used.

Reply:
I would rather have changes in a lot of places in this PR, than expose an API that doesn't make sense.
To me, it looks like using the variant interface/type in this package is not needed, so we should remove it.

Reply:
I think that the variant interface provides a safer design with the exposed variants, but I can update it to a string.

Reply:
My issue with the variant interface (which later gets converted to a `variant.Variant` type is, that this only _really_ makes sense in the context of Constellation.
If we expose this package as "public API", then it doesn't make sense to keep this restriction, but hide it behind some named interfaces.
This is especially confusing when other projects also provide their own `Variant` types, that might (or might not) fit into here.

If we just use a string, one can provide whatever they want. If the object does not exist, an error message is returned.
As already mentioned, we can provide the supported values as constants.